### ISSUE TITLE
BLW Error: NodeIP does not support data loss protection

### ISSUE THREAD

**Issue by khashmeshab** [lightningnetwork/lnd#4942]
### Background

Hi. I've recently installed lnd-v0.12.0-beta-rc6 on my amd64 VPS. It is configured to use Neutrino as its backend and I managed to finish the installation. I could open a channel using Electrum to it and also opened a channel from it to an ACINQ node. Then I tried to open a channel to it using an Android app, but both BLW and Eclair failed to open a channel. BLW said: "185.211.56.104 does not support data loss protection" and Eclair said: "Failed to open channel:incompatible features". I changed the lnwire/features.go to make DLP optional but it made no difference. The same error is also shown in rc5 and rc1. Since both BLW and Eclair could open channels to other nodes, I thought maybe it's an lnd bug.

### Your environment

* version of `lnd` **0.12.0-beta-rc6**
* which operating system (`uname -a` on *Nix) **Linux alpha.ir.nodes.gheysari.com 5.9.0-1-amd64 #1 SMP Debian 5.9.1-1 (2020-10-17) x86_64 GNU/Linux**
* version of `btcd`, `bitcoind`, or other backend **Neutrino**
* any other relevant environment details

### Steps to reproduce

Compile and install lnd using the official manual. Use this lnd.conf:

[Application Options]
debuglevel=trace
maxpendingchannels=10
listen=0.0.0.0:9735
externalip=185.211.56.104
alias=nevo.link
debuglevel=critical
maxlogfiles=1
maxlogfilesize=10
accept-keysend=1
allow-circular-route=1
ignore-historical-gossip-filters=1
max-channel-fee-allocation=1.0
max-cltv-expiry=5000
maxpendingchannels=10
minchansize=100000
sync-freelist=1
stagger-initial-reconnect=1
tlsautorefresh=1
tlsdisableautofill=1

[Bitcoin]
bitcoin.active=1
bitcoin.mainnet=1
#bitcoin.node=bitcoind
bitcoin.node=neutrino
bitcoin.defaultchanconfs=2
bitcoin.feerate=1000
bitcoin.minhtlc=1

[Neutrino]
neutrino.addpeer=btcd0.lightning.engineering
neutrino.addpeer=btcd-mainnet.lightning.computer
neutrino.addpeer=mainnet1-btcd.zaphq.io
neutrino.addpeer=mainnet2-btcd.zaphq.io
neutrino.addpeer=mainnet3-btcd.zaphq.io
neutrino.addpeer=mainnet4-btcd.zaphq.io
neutrino.feeurl=https://nodes.lightning.computer/fees/v1/btc-fee-estimates.json

[protocol]
protocol.wumbo-channels=1

[routerrpc]
routerrpc.apriorihopprob=0.5
routerrpc.aprioriweight=0.75
routerrpc.attemptcost=10
routerrpc.maxmchistory=10000
routerrpc.minrtprob=0.005
routerrpc.penaltyhalflife=6h0m0s

[routing]
routing.assumechanvalid=1

Then start lnd and unlock it. Then connect to your node using Eclair or BLW on Android. You will see the error. Or you may see the error using my node: `02d8a1d22732b1e2097ddb074bc15b3d14234ea5a0655df2997a67255df5c74765@185.211.56.104:9735`
Note that BLW shows the error right after you scan the qrcode, but Eclair shows it after you enter the desired capacity and try to open the channel.

### Expected behaviour

They should successfully open a channel.

### Actual behaviour

They show errors:
**BLW:** 185.211.56.104 does not support data loss protection
**Eclair:** Failed to open channel:incompatible features

Thank you.

**Comment by khashmeshab** [comment]
Output of `lncli getinfo`:

```
{
    "version": "0.12.0-beta.rc1 commit=v0.12.0-beta.rc1",
    "commit_hash": "f9641cdc64a953b6126eef1aa7987f83a064e06a",
    "identity_pubkey": "02d8a1d22732b1e2097ddb074bc15b3d14234ea5a0655df2997a67255df5c74765",
    "alias": "nevo.link",
    "color": "#3399ff",
    "num_pending_channels": 0,
    "num_active_channels": 2,
    "num_inactive_channels": 0,
    "num_peers": 4,
    "block_height": 667126,
    "block_hash": "00000000000000000001fe541f7479f740daef18417f84f5daf63a9305e38add",
    "best_header_timestamp": "1611293426",
    "synced_to_chain": true,
    "synced_to_graph": true,
    "testnet": false,
    "chains": [
        {
            "chain": "bitcoin",
            "network": "mainnet"
        }
    ],
    "uris": [
        "02d8a1d22732b1e2097ddb074bc15b3d14234ea5a0655df2997a67255df5c74765@185.211.56.104:9735"
    ],
    "features": {
        "0": {
            "name": "data-loss-protect",
            "is_required": true,
            "is_known": true
        },
        "5": {
            "name": "upfront-shutdown-script",
            "is_required": false,
            "is_known": true
        },
        "7": {
            "name": "gossip-queries",
            "is_required": false,
            "is_known": true
        },
        "9": {
            "name": "tlv-onion",
            "is_required": false,
            "is_known": true
        },
        "12": {
            "name": "static-remote-key",
            "is_required": true,
            "is_known": true
        },
        "14": {
            "name": "payment-addr",
            "is_required": true,
            "is_known": true
        },
        "17": {
            "name": "multi-path-payments",
            "is_required": false,
            "is_known": true
        },
        "19": {
            "name": "wumbo-channels",
            "is_required": false,
            "is_known": true
        },
        "23": {
            "name": "anchors-zero-fee-htlc-tx",
            "is_required": false,
            "is_known": true
        }
    }
}
```

**Comment by Roasbeef** [comment]
The BLW client is no longer actively developed: https://github.com/btcontract/lnwallet/commits/master

Until development picks up again and they implement the "remote static key" channel type (makes recovery of funds in a disaster easier), then you may not be able to open channels to the node. 

I recommend switching to a client that's actively developed in order to be up to date with the latest security/safety improvements in the protocol.

**Comment by khashmeshab** [comment]
You seem to be in such hurry to close this issue after 5 days of complete silence that you haven't even read my report. First, BLW "was" deprecated and it's now active again. Second, I wrote that Eclair shows the same error message too. Third, I know that BLW and Eclair don't support "remote static key", but I said that I made this feature optional in lnd. Why does both BLW and Eclair successfully connect to any other node, including lnd nodes, but not mine?
